data_structures/src/chain.rs

```diff
@@ -99,6 +99,44 @@ pub struct Block {
     pub txns: Vec<Transaction>,
 }
 
+/// The error type for operations on a [`Block`](Block)
+#[derive(Debug, PartialEq)]
+pub enum BlockError {
+    /// Indicates the block has no transactions in it.
+    Empty,
+    /// Indicates the first transaction of the block is no mint.
+    NoMint,
+    /// Indicates there was an error validating the transactions.
+    Transaction(TransactionError),
+    /// Indicates that the total value created by the mint transaction
+    /// of the block, and the output value of the rest of the
+    /// transactions, plus the block reward, don't add up
+    MismatchedMintValue,
+}
+
+impl Block {
+    /// Check if the block is valid.
+    pub fn validate(&self, block_reward: u64, pool: &TransactionsPool) -> Result<(), BlockError> {
```

Comment:
Why `Result` and not `WitnetResult`?

Reply:
Meh, I forgot about it, I'll change it to `WtinetResult` although I kind of find it more cumbersome to use :(

Reply:
I changed the code in `chain.rs` to use [`failure`](https://rust-lang-nursery.github.io/failure/) directly, what do you think? IMO our `WitnetResult` is in the middle of the approach followed in vanilla Rust and the approach followed in `failure` crate, so we should use just one: the `failure` approach since it let us create functions that may fail with different errors without any additional work (see the example in `failure`'s documentation). If we do this then I could refactor the code we already have to remove the `WitnetResult`